lib/mode-signup/index.js

```diff
@@ -374,16 +386,19 @@ SignupPanel.prototype.valid = function() {
     var username_parsed = username_parser.exec(username_input.val().toLowerCase());
 
     if (username_empty) {
+      widget.emit('signup error', new ValidationError('username empty'));
       widget._focusError(username_input);
       ok = false;
     }
     if (!username_parsed && !username_empty) {
+      widget.emit('signup error', new ValidationError('username invalid'));
       widget._focusError(username_input, widget.options.i18n.t('invalid'));
       ok = false;
     }
   }
 
   if (password_empty) {
+    widget.emit('signup error', new ValidationError('password empty'));
```

Comment:
same as signin and reset, save the error and emit at the end since we are not returning fast in either `.validate` method.